src/contracts/atlas/GasAccounting.sol

```diff
@@ -16,10 +16,241 @@ abstract contract GasAccounting is SafetyLocks {
     uint256 constant public BUNDLER_PREMIUM = 110; // the amount over cost that bundlers get paid
     uint256 constant public BUNDLER_BASE = 100;
 
+    uint256 constant private _ledgerLength = 6; // uint256(type(GasParty).max); // 6
+    Ledger[_ledgerLength] public ledgers;
+    AtlasLedger public atlasLedger;
+
     GasDonation[] internal _donations;
     AccountingData internal _accData;
 
-    constructor(address _simulator) SafetyLocks(_simulator) {}
+    constructor(address _simulator) SafetyLocks(_simulator) {
+        for (uint256 i; i < _ledgerLength; i++) {
+            ledgers[i].ledgerStatus = LedgerStatus.Inactive; // init the storage vars
+        }
+    }
+
+    function _initialAccounting() internal {
+        // Note: assumes msg.sender == tx.origin
+        if (msg.value > 0) {
+            uint64 bundlerDeposit = uint64(msg.value / tx.gasprice);
+            ledgers[uint256(GasParty.Bundler)] = Ledger({
+                deposited: bundlerDeposit,
+                withdrawn: 0,
+                unfulfilled: 0,
+                ledgerStatus: LedgerStatus.Surplus
+            });
+
+            atlasLedger = AtlasLedger({
+                totalBorrowed: 0,
+                totalDeposited: 0,
+                totalRequested: 0,
+                totalFulfilled: 0
+            });
+
+        } else {
+            uint64 bundlerDeposit = uint64(msg.value / tx.gasprice);
+            atlasLedger = AtlasLedger({
+                totalBorrowed: 0,
+                totalDeposited: bundlerDeposit,
+                totalRequested: 0,
+                totalFulfilled: bundlerDeposit
+            });
+        }
+    }
+
+    // NOTE: donations are simply deposits that have a different msg.sender than receiving party
+    function _deposit(GasParty party, uint256 amt) internal returns (uint256 balanceOwed) {
+
+        uint64 depositAmount = uint64(amt / tx.gasprice);
+        uint256 partyIndex = uint256(party);
+
+        Ledger memory pLedger = ledgers[partyIndex];
+        AtlasLedger memory aLedger = atlasLedger;
+
+        if (pLedger.unfulfilled != 0) {
+            if (pLedger.unfulfilled > depositAmount) {
+                pLedger.unfulfilled -= depositAmount;
+                aLedger.totalFulfilled += depositAmount;
```

Comment:
```suggestion
                aLedger.totalFulfilled += depositAmount;
                aLedger.totalDeposited += depositAmount;
```
Wouldn't we also need to increase aLedger totalDeposited here?

Reply:
I think the original intent here was to separate a cross-party loan (requested -> fulfilled) with an actual deposit. 

Not sure that I followed that idea al the way through. I'm actually wondering if we could skip the AtlasLedger storage struct all together.